torch/csrc/generic/StorageSharing.cpp

```diff
@@ -233,9 +248,38 @@ static PyObject * THPStorage_(shareCuda)(THPStorage *self)
 
     _handle = PyBytes_FromStringAndSize((char *)&handle, CUDA_IPC_HANDLE_SIZE);
     _offset_bytes = PyLong_FromSsize_t((Py_ssize_t)offset_bytes);
+
+    if (!torch::CudaIPCHaveRefCounter())
```

Comment:
Something I found surprising about this block, is that it appears to be initializing global state, but (1) the initialization code lives here (couldn't in principle other places need to initialize?) and (2) there's no locking of any sort (in this case, the GIL saves you, but there's nothing here that is Python-specific, no)?

Reply:
Aha, moved.